src/OpenTelemetry/Metrics/Reader/PeriodicExportingMetricReader.cs

```diff
@@ -46,45 +69,30 @@ public PeriodicExportingMetricReader(
 
         this.ExportIntervalMilliseconds = exportIntervalMilliseconds;
         this.ExportTimeoutMilliseconds = exportTimeoutMilliseconds;
+        this.useThreads = options.UseThreads;
 
-        this.exporterThread = new Thread(new ThreadStart(this.ExporterProc))
-        {
-            IsBackground = true,
-#pragma warning disable CA1062 // Validate arguments of public methods - needed for netstandard2.1
-            Name = $"OpenTelemetry-{nameof(PeriodicExportingMetricReader)}-{exporter.GetType().Name}",
-#pragma warning restore CA1062 // Validate arguments of public methods - needed for netstandard2.1
-        };
-        this.exporterThread.Start();
+        this.worker = this.CreateWorker();
+        this.worker.Start();
     }
 
     /// <inheritdoc />
     protected override bool OnShutdown(int timeoutMilliseconds)
     {
-        var result = true;
-
-        try
-        {
-            this.shutdownTrigger.Set();
-        }
-        catch (ObjectDisposedException)
-        {
-            return false;
-        }
+        var result = this.worker.Shutdown(timeoutMilliseconds);
 
         if (timeoutMilliseconds == Timeout.Infinite)
         {
-            this.exporterThread.Join();
-            result = this.exporter.Shutdown() && result;
+            return this.exporter.Shutdown() && result;
         }
-        else
+
+        if (timeoutMilliseconds == 0)
         {
-            var sw = Stopwatch.StartNew();
-            result = this.exporterThread.Join(timeoutMilliseconds) && result;
-            var timeout = timeoutMilliseconds - sw.ElapsedMilliseconds;
-            result = this.exporter.Shutdown((int)Math.Max(timeout, 0)) && result;
+            return this.exporter.Shutdown(0) && result;
         }
 
-        return result;
+        var sw = Stopwatch.StartNew();
+        var timeout = timeoutMilliseconds - sw.ElapsedMilliseconds;
```

Comment:
This code looks redundant now. There's no task happening between the stopwatch starting and timeout calculation now, so it should just be `this.exporter.Shutdown(timeout) && result`?